core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeExternalSorter.java

```diff
@@ -562,10 +562,6 @@ public UnsafeSorterIterator getSortedIterator() throws IOException {
     }
   }
 
-  @VisibleForTesting boolean hasSpaceForAnotherRecord() {
```

Comment:
Thank you for making a PR, but I'd not remove this. Although this is not used currently, this was added once inevitably. I feel this removal might be too hasty decision because `UnsafeExternalSorter` is designed to be based on `UnsafeInMemorySorter` and `UnsafeInMemorySorter.hasSpaceForAnotherRecord()` is a useful public method. It still looks okay to me to have this `VisibleForTesting` method to expose the underlying structure.